lib/Resque.php

```diff
@@ -52,44 +64,47 @@ public static function setBackend($server, $database = 0)
 	 */
 	public static function redis()
 	{
-		if (self::$redis !== null) {
-			return self::$redis;
+		// Detect when the PID of the current process has changed (from a fork, etc)
+		// and force a reconnect to redis.
+		$pid = getmypid();
+		if (self::$pid !== $pid) {
+			self::$redis = null;
+			self::$pid   = $pid;
 		}
 
-		if (is_callable(self::$redisServer)) {
-			self::$redis = call_user_func(self::$redisServer, self::$redisDatabase);
-		} else {
-			self::$redis = new Resque_Redis(self::$redisServer, self::$redisDatabase);
+		if(!is_null(self::$redis)) {
+			return self::$redis;
 		}
 
-		return self::$redis;
-	}
-
-	/**
-	 * fork() helper method for php-resque that handles issues PHP socket
-	 * and phpredis have with passing around sockets between child/parent
-	 * processes.
-	 *
-	 * Will close connection to Redis before forking.
-	 *
-	 * @return int Return vars as per pcntl_fork(). False if pcntl_fork is unavailable
-	 */
-	public static function fork()
-	{
-		if(!function_exists('pcntl_fork')) {
-			return false;
+		$server = self::$redisServer;
+		if (empty($server)) {
+			$server = 'localhost:6379';
 		}
 
-		// Close the connection to Redis before forking.
-		// This is a workaround for issues phpredis has.
-		self::$redis = null;
-
-		$pid = pcntl_fork();
-		if($pid === -1) {
-			throw new RuntimeException('Unable to fork child worker.');
+		if(is_array($server)) {
+			require_once dirname(__FILE__) . '/Resque/RedisCluster.php';
+			self::$redis = new Resque_RedisCluster($server);
+			if (empty(self::$auth) === false) {
+                self::$redis->auth(self::$auth);
+            }
+		}
+		else {
+			if (strpos($server, 'unix:') === false) {
+				list($host, $port) = explode(':', $server);
+			}
+			else {
+				$host = $server;
+				$port = null;
+			}
+			require_once dirname(__FILE__) . '/Resque/Redis.php';
+			self::$redis = new Resque_Redis($host, $port);
+            if (empty(self::$auth) === false) {
+                self::$redis->auth(self::$auth);
+            }
 		}
 
-		return $pid;
+		self::$redis->select(self::$redisDatabase);
+		return self::$redis;
```

Comment:
All of this logic is unnecessary with Credis, and is in fact redundant.  Assuming it dates from before the switch to Composer, as well.